lte/gateway/c/oai/tasks/amf/amf_recv.cpp

```diff
@@ -0,0 +1,292 @@
+     * accept message
+     * TODO Note:currently adapting the way
+     */
+    amf_ctx_guti = (guti_m5_t*) &amf_guti;
+
+    /* store this GUTI in
+     * unordered_map<imsi64_t, guti_and_amf_id_t> amf_supi_guti_map
+     */
+    imsi64_t imsi64                = amf_imsi_to_imsi64(&imsi);
+    guti_and_amf_id.amf_guti       = amf_guti;
+    guti_and_amf_id.amf_ue_ngap_id = ue_id;
+
+    if (amf_supi_guti_map.size() == 0) {
+      // first entry.
+      amf_supi_guti_map.insert(
+          std::pair<imsi64_t, guti_and_amf_id_t>(imsi64, guti_and_amf_id));
+    } else {
+      /* already elements exist then check if same imsi already present
+       * if same imsi then update/overwrite the element
+       */
+      std::unordered_map<imsi64_t, guti_and_amf_id_t>::iterator found_imsi =
+          amf_supi_guti_map.find(imsi64);
+      if (found_imsi == amf_supi_guti_map.end()) {
+        // it is new entry to map
+        amf_supi_guti_map.insert(
+            std::pair<imsi64_t, guti_and_amf_id_t>(imsi64, guti_and_amf_id));
+      } else {
+        // Overwrite the second element.
+        found_imsi->second = guti_and_amf_id;
+      }
+    }
+  }
+  /*
+   * Execute the identification completion procedure
+   */
+  rc = amf_proc_identification_complete(
+      ue_id, p_imsi, p_imei, p_imeisv, (uint32_t*) (p_tmsi), amf_ctx_guti);
+  OAILOG_FUNC_RETURN(LOG_NAS_EMM, rc);
+}
+
+/****************************************************************************
+ **                                                                        **
+ ** Name:    amf_handle_authentication_response()                          **
+ **                                                                        **
+ ** Description: Processes Authentication Response message                 **
+ **                                                                        **
+ ** Inputs:  ue_id:      UE lower layer identifier                         **
+ **      msg:       The received AMF message                               **
+ **      Others:    None                                                   **
+ **                                                                        **
+ ** Outputs:     amf_cause: AMF cause code                                 **
+ **      Return:    RETURNok, RETURNerror                                  **
+ **      Others:    None                                                   **
+ **                                                                        **
+ ***************************************************************************/
+int amf_handle_authentication_response(
+    amf_ue_ngap_id_t ue_id, AuthenticationResponseMsg* msg, int amf_cause,
+    amf_nas_message_decode_status_t status) {
+  OAILOG_FUNC_IN(LOG_NAS_AMF);
+  int rc = RETURNok;
+  OAILOG_DEBUG(LOG_NAS_AMF, "Received AUTHENTICATION_RESPONSE message\n");
+  /*
+   * Message checking
+   */
+  if (NULL == msg->autn_response_parameter.response_parameter) {
```

Comment:
Put macro on the right hand side
```suggestion
  if (msg->autn_response_parameter.response_parameter == NULL) {
```

Reply:
Fixed